client/dashboard/sites/settings-subscription-gifting/index.tsx

```diff
@@ -33,21 +35,20 @@ const form = {
 	fields,
 };
 
-export function getSubscriptionGiftingSettingBadges( settings: SiteSettings ) {
-	return settings.wpcom_gifting_subscription
-		? [ { text: __( 'Enabled' ), intent: 'success' as const } ]
-		: [ { text: __( 'Disabled' ) } ];
-}
-
 export default function SubscriptionGiftingSettings() {
 	const { siteSlug } = siteSettingsSubscriptionGiftingRoute.useParams();
+	const { data: siteData } = useQuery( siteQuery( siteSlug ) );
 	const { data } = useQuery( siteSettingsQuery( siteSlug ) );
 	const mutation = useMutation( siteSettingsMutation( siteSlug ) );
 
-	if ( ! data ) {
+	if ( ! data || ! siteData ) {
 		return null;
 	}
 
+	if ( ! hasSubscriptionGiftingFeature( siteData.site ) ) {
```

Comment:
I think we have a flaw in `siteQuery` at the moment. It's not really clear to me why `siteQuery` regroups the "site" endpoint and all the other endpoints that are needed for the site overview. It seems these should be separate queries.

Yes, they all need to be in the "loader" for "site overview", but they don't need to be in the loaders of all the sub routes.

cc @ntsekouras 

(I realize this is out of scope of this PR, but let's try to work on this on a follow-up before it gets out of hand)

Reply:
OK, I can work something out in a separate PR.

Then do we really need to have separate "query" vs "API call"? It seems to me that with this, we don't want "grouping" but instead a 1:1 mapping between query and API call, so why do we need to separate them.

Reply:
I mean for example we can just export this:

```ts
export function siteSettingsQuery( siteId: string ) {
	return {
		queryKey: [ 'site-settings', siteId ],
		queryFn: () => {
			const { settings } = await wpcom.req.get( {
				path: `/sites/${ siteIdOrSlug }/settings`,
				apiVersion: '1.4',
			} );
			return settings;
		},
	};
}
```

(we "inline" the API call here)

Reply:
> Then do we really need to have separate "query" vs "API call"? It seems to me that with this, we don't want "grouping" but instead a 1:1 mapping between query and API call, so why do we need to separate them.

- "query" is tied to the application layer.
- "api call" is tied to the data layer.

I agree they feel like duplication for now. If I had to eliminate one, I'd remove the queries but it would create some duplication between loaders and components (query keys and extra query cache configuration...)

Reply:
What's the downside of inlining the API calls as I demonstrated above? There's nothing that consumes this data layer other than queries for now.

(Just curious, not that something that we need to do immediately)

Reply:
I like for the data layer to be separate from the implementation of the app for several reasons:

 - It actually defines the "model" of our API clearly (the types)
 - It can later be made "adaptable". By that I mean the same application (hosting dashboard) can be made to work with alternative REST API layers, by having an "adapter" that swaps the data layer while keeping the types and the queries the same. Think maybe using the same hosting dashboard in pressable / .com / wpcloud...

Reply:
>Yes, they all need to be in the "loader" for "site overview", but they don't need to be in the loaders of all the sub routes.

@youknowriad for clarification, your suggestion would be to have an `siteOverviewQuery` like the current `siteQuery` and break down the rest? For example `siteQuery` should just `fetchSite`. Is this correct?

Reply:
no, my suggestion is to have a "query" for each part of the current siteQuery instead of a single query with all the data.